wire-ios/Wire-iOS/Sources/UserInterface/Conversation/Create/Cells/ConversationCreateNameCell.swift

```diff
@@ -32,15 +44,56 @@ final class ConversationCreateNameCell: UICollectionViewCell {
         fatalError("init?(coder aDecoder: NSCoder) is not implemented")
     }
 
-    fileprivate func setup() {
-        textField.translatesAutoresizingMaskIntoConstraints = false
-        textField.isAccessibilityElement = true
-        textField.accessibilityIdentifier = "textfield.newgroup.name"
-        textField.placeholder = L10n.Localizable.Conversation.Create.GroupName.placeholder.localizedUppercase
-        textField.accessibilityValue = L10n.Accessibility.CreateConversation.SearchView.description
+    private func setup() {
+        setupStackView()
+        setupGroupNameLabel()
+        setupTextField()
+        setupConstraints()
+        setupNotifications()
+    }
+
+    private func setupStackView() {
+        stackView.axis = .vertical
+        stackView.spacing = 8
+        stackView.alignment = .fill
+        stackView.distribution = .fill
+        stackView.translatesAutoresizingMaskIntoConstraints = false
+        contentView.addSubview(stackView)
+    }
+
+    private func setupGroupNameLabel() {
+        stackView.addArrangedSubview(groupNameLabel)
+    }
+
+    private func setupTextField() {
+        textField.borderStyle = .roundedRect
+        textField.layer.borderColor = SemanticColors.SearchBar.borderInputView.cgColor
+        textField.font = .font(for: .body1)
+        stackView.addArrangedSubview(textField)
+    }
+
+    private func setupConstraints() {
+        NSLayoutConstraint.activate([
+            stackView.topAnchor.constraint(equalTo: contentView.topAnchor),
+            stackView.leadingAnchor.constraint(equalTo: contentView.leadingAnchor, constant: 16),
+            stackView.trailingAnchor.constraint(equalTo: contentView.trailingAnchor, constant: -16),
+            stackView.bottomAnchor.constraint(lessThanOrEqualTo: contentView.bottomAnchor),
+
+            textField.heightAnchor.constraint(equalToConstant: 46)
+        ])
+    }
 
-        contentView.addSubview(textField)
-        textField.fitIn(view: contentView)
+    private func setupNotifications() {
+        NotificationCenter.default.publisher(for: UITextField.textDidBeginEditingNotification, object: textField)
+            .sink { [weak self] _ in
+                self?.groupNameLabel.textColor = UIColor.accent()
+            }
+            .store(in: &cancellables)
 
+        NotificationCenter.default.publisher(for: UITextField.textDidEndEditingNotification, object: textField)
+            .sink { [weak self] _ in
+                self?.groupNameLabel.textColor = SemanticColors.Label.textUserPropertyCellName
+            }
+            .store(in: &cancellables)
```

Comment:
@jullianm I know that you're familiar with Combine. Let me know what you think about setting up the notifications this way. I wanted to avoid using the addObserver with the `#selector` pattern. And that's why I used Combine.

Reply:
Looks good! that's something we can consider reuse across the project so I'd suggest we could introduce UITextField extension helper methods, like so: 

```
extension UITextField {
    var textDidChangePublisher: AnyPublisher<String, Never> {
        NotificationCenter.default.publisher(
            for: UITextField.textDidChangeNotification,
            object: self
        )
        .compactMap { ($0.object as? UITextField)?.text }
        .eraseToAnyPublisher()
    }
}
```

Simply at callsite: 

```
textField.textDidChangePublisher
    .sink { text in
        // do something
    }
```

Reply:
This is a good idea. I would tackle it on a separate PR since it's going to touch multiple files/textfields.